pallets/runtime/tests/src/contract_test.rs

```diff
@@ -154,45 +158,55 @@ where
     T::Hashing::hash(&b"abc".encode())
 }
 
+/// Creates a `TestExternalities` using the given `network_fee_share` and `protocol_base_fees`.
+///
+/// It also enables the `put_code` extrinsics.
+fn make_externalities(
+    network_fee_share: Perbill,
+    protocol_base_fees: MockProtocolBaseFees,
+) -> TestExternalities {
+    ExtBuilder::default()
+        .network_fee_share(network_fee_share)
+        .set_protocol_base_fees(protocol_base_fees)
+        .set_contracts_put_code(true)
+        .build()
+}
+
 #[test]
 fn check_put_code_functionality() {
     // Build wasm and get code_hash
     let (wasm, code_hash) = compile_module::<TestStorage>("flipper").unwrap();
     let protocol_fee = MockProtocolBaseFees(vec![(ProtocolOp::ContractsPutCode, 500)]);
 
-    ExtBuilder::default()
-        .network_fee_share(Perbill::from_percent(0))
-        .set_protocol_base_fees(protocol_fee)
-        .build()
-        .execute_with(|| {
-            let alice = AccountKeyring::Alice.public();
-            // Create Alice account & the identity for her.
-            let (_, alice_did) = make_account_without_cdd(alice).unwrap();
+    make_externalities(Perbill::from_percent(0), protocol_fee).execute_with(|| {
```

Comment:
This bit is still duplicated all over, so let's make a higher order function, called with `wasm, code_hash` as inputs, and use it to dedup.